pkg/kafka/auth/config.go

```diff
@@ -43,7 +43,8 @@ func (config *AuthenticationConfig) SetConfiguration(saramaConfig *sarama.Config
 	if strings.Trim(authentication, " ") == "" {
 		authentication = none
 	}
-	if config.Authentication == tls {
+
+	if config.Authentication == tls || (!config.TLS.Insecure && config.TLS.ServerName != "") {
```

Comment:
in the 2nd OR clause, why isn't it enough to only check for `!config.TLS.Insecure`?